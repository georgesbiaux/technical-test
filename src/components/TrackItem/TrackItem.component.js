import React from 'react';
import {
  TrackItemFooter,
  TrackItemUserIcon,
  TrackItemInfo,
  TrackItemPriorityOverlay,
} from 'components';
import StyledTrackItem from './TrackItem.style';

const TrackItem = props => (
  <StyledTrackItem className="track-item">
    <TrackItemUserIcon pictureUrl={props.track.adder.pictureUrl} />
    <div className="track-container">
      <div className="track-row">
        <TrackItemInfo track={props.track} />
      </div>
      <TrackItemFooter track={props.track} />
    </div>
    {props.track.priority && <TrackItemPriorityOverlay />}
  </StyledTrackItem>
);

export default TrackItem;
