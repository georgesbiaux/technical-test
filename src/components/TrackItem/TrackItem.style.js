import styled from 'styled-components';

const StyledTrackItem = styled.div`
  background-color: #fbfbfb;
  border-bottom: 1px solid #eeeeee;
  display: flex;
  flex-direction: row;

  .user-icon {
    height: 60px;
    width: 60px;
    margin: 10px;
    border-radius: 50%;
  }

  .track-container {
    flex: 1;
    display: flex;
    flex-direction: column;
  }

  .track-row {
    flex: 1;
    padding-top: 8px;
    display: flex;
    flex-direction: row;
  }

  .track-item-button {
    flex: 1;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
  }
`;

export default StyledTrackItem;
