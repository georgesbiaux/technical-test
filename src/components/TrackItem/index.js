export { default } from './TrackItem.component';
