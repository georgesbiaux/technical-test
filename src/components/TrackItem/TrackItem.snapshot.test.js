import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TrackItem from './TrackItem.component';

describe('<TrackItem>', () => {
  it('should render correctly when priority is false', () => {
    const props = {

      track: {
        id: 1,
        name: 'Beast Of Burden',
        duration: 211,
        priority: false,
        artist: 'The Rolling Stones',
        adder: {
          id: 1,
          name: 'kant',
          pictureUrl:
            'https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd',
        },
        pictureUrl:
          'https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372',
        votes: {
          count: 0,
          userVoted: false,
        },
      },
    };
    const tree = shallow(<TrackItem {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });

  it('should render correctly when priority is true', () => {
    const props = {
      track: {
        id: 1,
        name: 'Beast Of Burden',
        duration: 211,
        priority: true,
        artist: 'The Rolling Stones',
        adder: {
          id: 1,
          name: 'kant',
          pictureUrl:
            'https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd',
        },
        pictureUrl:
          'https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372',
        votes: {
          count: 0,
          userVoted: false,
        },
      },
    };
    const tree = shallow(<TrackItem {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
