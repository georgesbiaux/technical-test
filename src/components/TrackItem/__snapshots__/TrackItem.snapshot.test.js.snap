// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackItem> should render correctly when priority is false 1`] = `
<styled.div
  className="track-item"
>
  <styled.div
    pictureUrl="https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd"
  />
  <div
    className="track-container"
  >
    <div
      className="track-row"
    >
      <TrackItemInfo
        track={
          Object {
            "adder": Object {
              "id": 1,
              "name": "kant",
              "pictureUrl": "https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd",
            },
            "artist": "The Rolling Stones",
            "duration": 211,
            "id": 1,
            "name": "Beast Of Burden",
            "pictureUrl": "https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372",
            "priority": false,
            "votes": Object {
              "count": 0,
              "userVoted": false,
            },
          }
        }
      />
      <Connect(Component)
        track={
          Object {
            "adder": Object {
              "id": 1,
              "name": "kant",
              "pictureUrl": "https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd",
            },
            "artist": "The Rolling Stones",
            "duration": 211,
            "id": 1,
            "name": "Beast Of Burden",
            "pictureUrl": "https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372",
            "priority": false,
            "votes": Object {
              "count": 0,
              "userVoted": false,
            },
          }
        }
      />
    </div>
    <TrackItemFooter
      track={
        Object {
          "adder": Object {
            "id": 1,
            "name": "kant",
            "pictureUrl": "https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd",
          },
          "artist": "The Rolling Stones",
          "duration": 211,
          "id": 1,
          "name": "Beast Of Burden",
          "pictureUrl": "https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372",
          "priority": false,
          "votes": Object {
            "count": 0,
            "userVoted": false,
          },
        }
      }
    />
  </div>
</styled.div>
`;

exports[`<TrackItem> should render correctly when priority is true 1`] = `
<styled.div
  className="track-item"
>
  <styled.div
    pictureUrl="https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd"
  />
  <div
    className="track-container"
  >
    <div
      className="track-row"
    >
      <TrackItemInfo
        track={
          Object {
            "adder": Object {
              "id": 1,
              "name": "kant",
              "pictureUrl": "https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd",
            },
            "artist": "The Rolling Stones",
            "duration": 211,
            "id": 1,
            "name": "Beast Of Burden",
            "pictureUrl": "https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372",
            "priority": true,
            "votes": Object {
              "count": 0,
              "userVoted": false,
            },
          }
        }
      />
      <Connect(Component)
        track={
          Object {
            "adder": Object {
              "id": 1,
              "name": "kant",
              "pictureUrl": "https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd",
            },
            "artist": "The Rolling Stones",
            "duration": 211,
            "id": 1,
            "name": "Beast Of Burden",
            "pictureUrl": "https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372",
            "priority": true,
            "votes": Object {
              "count": 0,
              "userVoted": false,
            },
          }
        }
      />
    </div>
    <TrackItemFooter
      track={
        Object {
          "adder": Object {
            "id": 1,
            "name": "kant",
            "pictureUrl": "https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd",
          },
          "artist": "The Rolling Stones",
          "duration": 211,
          "id": 1,
          "name": "Beast Of Burden",
          "pictureUrl": "https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372",
          "priority": true,
          "votes": Object {
            "count": 0,
            "userVoted": false,
          },
        }
      }
    />
  </div>
  <TrackItemPriorityOverlay />
</styled.div>
`;
